src/functionaltests/java/com/ericsson/ei/utils/JenkinsManager.java

```diff
@@ -0,0 +1,226 @@
+     * @return
+     * @throws URISyntaxException
+     */
+    public boolean jobHasBeenTriggered(String jobName) throws URISyntaxException {
+        boolean triggered = false;
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.GET);
+        if (jobName == null) {
+            jobName = DEFAULT_JOB_NAME;
+        }
+
+        String endpoint = "/job/" + jobName + "/1/api/json";
+
+        httpRequest.setHost(host).setPort(port).addHeader("Authorization", "Basic " + encoding)
+                .addHeader("Content-type", MediaType.APPLICATION_JSON_VALUE)
+                .setEndpoint(endpoint);
+
+        ResponseEntity<String> response = httpRequest.performRequest();
+        triggered = response.getStatusCode() == HttpStatus.OK;
+        return triggered;
+    }
+
+    /**
+     *
+     * This function recieves a jenkins job name and deletes that job from the
+     * jenkins system
+     *
+     * @param jobName
+     * @return
+     * @throws URISyntaxException
+     */
+    public boolean deleteJob(String jobName) throws URISyntaxException {
+        boolean deleted = false;
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.POST);
+        if (jobName == null) {
+            jobName = DEFAULT_JOB_NAME;
+        }
+
+        String endpoint = "/job/" + jobName + "/doDelete";
+
+        httpRequest.setHost(host).setPort(port).addHeader("Authorization", "Basic " + encoding)
+                .addHeader("Content-type", MediaType.APPLICATION_JSON_VALUE).addHeader("Jenkins-Crumb", crumb)
+                .setEndpoint(endpoint);
+
+        ResponseEntity<String> response = httpRequest.performRequest();
+        deleted = response.getStatusCode() == HttpStatus.FOUND;
+        return deleted;
+    }
+
+    /**
+     * Gets the jenkins crumb if any
+     *
+     * @return
+     * @throws URISyntaxException
+     */
+    private String getCrumb() throws URISyntaxException {
+        String crumb = "";
+        HttpRequest httpRequest = new HttpRequest(HttpMethod.GET);
+
+        httpRequest.setHost(host).setPort(port).addHeader("Authorization", "Basic " + encoding)
+                .addHeader("Content-type", MediaType.APPLICATION_JSON_VALUE).setEndpoint("/crumbIssuer/api/json");
+
+        ResponseEntity<String> response = httpRequest.performRequest();
+        boolean success = response.getStatusCode() == HttpStatus.OK;
+
+        if (success) {
+            JSONObject jsonObj = new JSONObject(response.getBody());
+            crumb = jsonObj.getString("crumb");
+        }
+        return crumb;
+    }
+
+    private static String getStringFromFile(String filepath) throws IOException {
+        return new String(Files.readAllBytes(Paths.get(filepath)), StandardCharsets.UTF_8);
+    }
+
+    /**
+     * Replaces tags in string with valid information.
+     *
+     * @param text
+     * @param token
+     * @param script
+     * @return
+     */
+    private String stringReplaceText(String text, String token, String script) {
```

Comment:
This function could be made more general and could be used in future by other needs:
"private String stringReplaceText(String text, String textToBeMatched, String textToBeInserted) {
return text.replaceAll(textMatchToBeMatched, textToBeInserted);
}"
Or something similar.
And put the function in some utils lib class.

Reply:
Maybe could be possible.

Reply:
Isn't that just an unnecessary wrapper Tobias? Does "text.replaceAll" do exactly the same thing as  "stringReplaceText" would do?

Reply:
Erik is correct.
This is not a generic function it is just the replaceAll moved to it´s own function for better structure.

Reply:
If so, then function name and java doc should be more specific and not general described.

Reply:
Agreed